.gitlab-ci-check-docker-build.yml

```diff
@@ -191,3 +191,38 @@ trigger:saas:sync-staging-component:
           -F variables[SYNC_IMAGE_TAG]=${DOCKER_PUBLISH_COMMIT_TAG}
           https://gitlab.com/api/v4/projects/17809112/trigger/pipeline
     - done
+
+# saas-specific job to retag docker images after saas-* tags are pushed to the repository
+publish:image:saas:
+  stage: publish
+  only:
+    refs:
+      - /^saas-[a-zA-Z0-9.]+$/
+  tags:
+    - docker
+  image: docker
+  services:
+    - docker:19.03.5-dind
+  dependencies:
+    - build:docker
```

Comment:
In the same pull request you are adding:

```diff
 build:docker:
   stage: build
+  except:
+    - /^saas-[a-zA-Z0-9.]+$/
```

Will that work? IIRC, jobs that have matching exceptions will just lead to no dependency, IOW the publish job will still run, but the build will not.

Reply:
> I assume that this removal of testing is because all of this is expected to have gone through `staging` already?

Yes, the idea is that pushing a tag with the `saas-` prefix should only re-tag to already built and pushed docker image (staging_SHA). I don't know if this is going to work, I've tried my best but I don't know how to actually test the updated pipeline. @lluiscampos?

Reply:
@tranchitella Kristian is right in that this job cannot depend on `build:docker`. I should not have any dependency as this will just docker pull + docker tag + docker push (as you have correctly done).

Reply:
Done!